packages/core/src/libraries/hook/index.ts

```diff
@@ -188,6 +190,48 @@ export const createHookLibrary = (queries: Queries) => {
     await sendWebhooks(webhooks, consoleLog);
   };
 
+  /**
+   * Trigger data hooks with the given data mutation context. All context objects will be used to trigger hooks.
+   */
+  const triggerExceptionHooks = async (
+    consoleLog: ConsoleLog,
+    contextManager: ExceptionHookContextManager
+  ) => {
+    if (contextManager.contextArray.length === 0) {
+      return;
+    }
+
+    const found = await findAllHooks();
+
+    // Fetch application detail if available
+    const { applicationId } = contextManager.metadata;
+    const application = applicationId
+      ? await trySafe(async () => findApplicationById(applicationId))
+      : undefined;
```

Comment:
Should we filter the available webhooks first? If the list is empty, we can skip querying the application data.

Reply:
I prefer not to include extra code changes into the PR, as it is irrelavent to this feature request.